redux/actions/preferencesActions.js

```diff
@@ -0,0 +1,55 @@
+import { 
```

Comment:
updatedauthor and other functions i can't see their functions, where they are being called. When you make those functions use a dispatcher